File: python/best_time_to_buy_and_sell_stock_ii.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        maxPrice: int = 0
        for i in range(len(prices) - 1):
            if prices[i + 1] < prices[i]:
                continue
            else:
                maxPrice += prices[i + 1] - prices[i]

        return maxPrice


def test_maxProfit_case_1():
    # arrange
    prices: list[int] = [7, 1, 5, 3, 6, 4]
    expected: int = 7

    # act
    solution = Solution()
    actual = solution.maxProfit(prices)

    # assert
    assert expected == actual


def test_maxProfit_case_2():
    # arrange
    prices: list[int] = [1, 2, 3, 4, 5]
    expected: int = 4

    # act
    solution = Solution()
    actual = solution.maxProfit(prices)

    # assert
    assert expected == actual


def test_maxProfit_case_3():
    # arrange
    prices: list[int] = [7, 6, 4, 3, 1]
    expected: int = 0

    # act
    solution = Solution()
    actual = solution.maxProfit(prices)

    # assert
    assert expected == actual

File: rust/src/code/best_time_to_buy_and_sell_stock_ii.rs
struct Solution {}

impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut max_price = 0;
        for i in 0..prices.len() - 1 {
            if prices[i + 1] < prices[i] {
                continue;
            }

            max_price += prices[i + 1] - prices[i];
        }

        max_price
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn max_profit_case_1() {
        // arrange
        let prices: Vec<i32> = vec![7, 1, 5, 3, 6, 4];
        let expected: i32 = 7;

        // act
        let actual = Solution::max_profit(prices);

        // assert
        assert_eq!(expected, actual);
    }
}
